### PR TITLE
Respect the preload-none option

### PR DESCRIPTION
**Pull request by bc-bbay**
Don’t start downloading segments if preload is set to “none”

### CODE REVIEW

File: src/videojs-hls.js [hunk 1]
```diff
@@ -558,6 +555,11 @@ videojs.Hls.prototype.fillBuffer = function(offset) {
     segment,
     segmentUri;
 
+  // if preload is set to "none", do not download segments until playback is requested
+  if (!player.hasClass('vjs-has-started') && player.options.preload === "none") {
```
Comment (dmlap) [inline src/videojs-hls.js]: Should the second condition be `player.preload() === 'none'`? `options` are set on player creation and (I believe) ignored from that point out.

**Comment by dmlap** [comment]
Nice!

**Comment by dmlap** [comment]
Related: #233

**Comment by shaharmor** [comment]
+100 ! this is very disturbing!

**Comment by dmlap** [comment]
@bc-bbay I just tried this out on a player and it appears preload() is (incorrectly) returning a boolean. Can you double check this is working as you expect?

**Comment by bc-bbay** [comment]
You're right, preload was a boolean in the SWF. I've updated that project and this pull now depends on this one in video-js-swf: https://github.com/videojs/video-js-swf/pull/155

**Comment by bclwhitaker** [comment]
LGTM

**Pull request merged**